src/file-box.ts

```diff
@@ -1017,14 +1017,12 @@ class FileBox implements Pipeable, FileBoxInterface {
   pipe<T extends Writable> (
     destination: T,
   ): T {
-    this.toStream().then(stream => {
-      stream.on('error', e => {
-        console.info('error:', e)
-
-        destination.emit('error', e)
+    this.toStream()
+      .then(stream => {
+        stream.once('error', e => destination.destroy(e))
+        return stream.pipe(destination)
       })
-      return stream.pipe(destination)
-    }).catch(e => destination.emit('error', e))
+      .catch(e => destination.destroy(e))
```

Comment:
I think `destination.emit('error')` will be enough.

What's the reason you want to destroy the destination at here?

Reply:
When an error occurs in the src stream, it means that the src stream is no longer available, and if just emit an error to the dest stream, dest will not be closed, and a memory leak may occur.